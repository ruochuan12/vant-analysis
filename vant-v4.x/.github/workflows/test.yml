name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - main

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4

    - name: Install pnpm
      run: corepack enable

    - uses: actions/setup-node@v3
      with:
        node-version: '18'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: macos-14
    steps:
    - uses: actions/checkout@v4

    - name: Install pnpm
      run: corepack enable

    - uses: actions/setup-node@v3
      with:
        node-version: '18'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: pnpm run test:coverage

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v3
      with:
        token: ${{ secrets.CODECOV_TOKEN }}
        directory: ./packages/vant/test/coverage

  build:
    runs-on: macos-14
    steps:
    - uses: actions/checkout@v4

    - name: Install pnpm
      run: corepack enable

    - uses: actions/setup-node@v3
      with:
        node-version: '18'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
