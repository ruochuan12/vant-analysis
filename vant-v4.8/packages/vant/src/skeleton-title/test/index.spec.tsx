import { mount } from '../../../test';
import { SkeletonTitle } from '..';

test('should skeleton title works', () => {
  const wrapper = mount(SkeletonTitle);

  expect(wrapper.html()).toMatchSnapshot();
});

test('should skeleton title works with props', () => {
  const wrapper = mount(SkeletonTitle, {
    props: {
      round: true,
      titleWidth: '200rem',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
