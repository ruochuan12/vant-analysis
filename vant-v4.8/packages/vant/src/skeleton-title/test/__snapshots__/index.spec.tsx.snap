// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should skeleton title works 1`] = `"<h3 class="van-skeleton-title"></h3>"`;

exports[`should skeleton title works with props 1`] = `"<h3 class="van-skeleton-title van-skeleton-title--round" style="width: 200rem;"></h3>"`;
